src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java

```diff
@@ -267,6 +307,80 @@ private static void verifyAstRaw(String expectedTextPrintFileName, String actual
                 "Generated AST from Java code should match pre-defined AST");
     }
 
+    private static final class AssertGeneratedJava14TextBlockLexer
```

Comment:
I would like to know a better way to reuse the existing code (see [here](https://github.com/checkstyle/checkstyle/blob/51836b52b04c5681a34254b09b78364a04be87dc/src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java#L270)) .  Should I make AssertGeneratedJavaLexer.java not final, then extend it, or should I refactor AssertGeneratedJavaLexer.java to work with the new lexer, too?

Reply:
This class is `final` simply because there was no reason not to make it `final` before. If you need to re-factor this to support the grammar for text blocks, then go ahead.

Reply:
Do you think it is better to factor out a superclass, or to create an interface?

Reply:
After re-examining this, since each class must be a descendant of it's respective lexer, and an interface for these seems like overkill, I am going to leave this as it is.